lib/players/player.rb

```diff
@@ -5,11 +5,15 @@ def initialize(behavior, token, user_interface)
     @user_interface = user_interface
   end
 
-  def take_turn(board, turns_remaining)
+  def take_turn(state, board, turns_remaining)
     grid = board.grid
     move = get_move(grid, turns_remaining)
-    add_move(board, move)
-    display_board(grid)
+    exit?(state, move)
```

Comment:
I am not crazy about this. It might just be the method name though. `exit?` doesn't really tell us what the method is doing. You know what I mean?

Reply:
Adjusted the method name to be `wants_to_exit?`.  Also debated `player_wants_to_exit?` but decided to go with the first one since it is in the player class.